### PR TITLE
Add Reducers and Actions

### PR DESCRIPTION
**Pull request by Metaverse-Nabeel**
### The following changes have been made to this PR:

- Added Redux Toolkit (npm install react-redux @reduxjs/toolkit).
- Created a directory that will contain all your Redux logic (/src/redux)
- Configured a Redux store (/src/redux/store.js)
- Defined a slice of state for books that:
- Included an array of books (initial state: empty array)
- Included a reducer that adds a book
- Included a reducer that removes a book
- Defined a slice of state for categories that:
- Included an array of categories (initial state: empty array)
- Included a reducer that checks the status and always returns "Under construction" (the initial state should check to that string)
- Structured application files using a "feature folder" approach and used the [ducks pattern](https://github.com/erikras/ducks-modular-redux) for your Redux files. 
- Updated Readme File
- Corrected Linters

### CODE REVIEW

File: src/redux/categories/categoriesSlice.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const categoriesReducer = createSlice({
+  name: 'categories',
+  initialState: [],
+  reducers: {
+    checkStatus() {
+      return 'Under construction';
+    },
```
Comment (GedeonTS) [inline src/redux/categories/categoriesSlice.js]: - According to [this requirement](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/add_reducers_an_actions_v2.md#:~:text=Includes%20a%20reducer%20that%20checks%20the%20status%20and%20always%20returns%20%22Under%20construction%22%20(the%20initial%20state%20should%20check%20to%20that%20string)), Kindly make sure the initial state check to the string Under Construction. it should be looking like this 
`checkStatus: (state, action) => {
      state =
        action.payload === "Under construction"
          ? "Under construction"
          : state
    },
`

File: src/redux/categories/categoriesSlice.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  },
+});
+
+export const { checkStatus } = categoriesReducer.actions;
+export default categoriesReducer;
```
Comment (GedeonTS) [inline src/redux/categories/categoriesSlice.js]: - Kindly make sure to export the reducer, like `export default categoriesReducer.reducer;`

File: src/redux/books/booksSlice.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const booksReducer = createSlice({
```
Comment (GedeonTS) [inline src/redux/books/booksSlice.js]: - [OPTIONAL] I highly recommend you to change the name of your Slice to something like `booksSlice` to give it more sense, as it is not the reducer

File: src/redux/categories/categoriesSlice.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const categoriesReducer = createSlice({
```
Comment (GedeonTS) [inline src/redux/categories/categoriesSlice.js]: - [OPTIONAL] I highly recomment you to change the name of your Slice to something like `CategoriesSlice` to give it more sense, as it is not the reducer

**Review by GedeonTS — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED :recycle: :

Hi @Metaverse-Nabeel , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights :tophat:  :+1: 
- Good PR Description :+1: 
- No Linter Errors :100: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/categories/categoriesSlice.js (hunk 1 above)
Comment (Metaverse-Nabeel) [reply src/redux/categories/categoriesSlice.js]: Hello @GedeonTS 

Thank you for your kind feedback. I shall make the changes and ask for a review again.

Regards

File: src/redux/categories/categoriesSlice.js (hunk 2 above)
Comment (Metaverse-Nabeel) [reply src/redux/categories/categoriesSlice.js]: Hello @GedeonTS 

Thank you for your kind feedback. I shall make the changes and ask for a review again.

Regards

File: src/redux/books/booksSlice.js (hunk 3 above)
Comment (Metaverse-Nabeel) [reply src/redux/books/booksSlice.js]: Hello @GedeonTS 

Thank you for your kind feedback. I shall make the changes and ask for a review again.

Regards

File: src/redux/categories/categoriesSlice.js (hunk 4 above)
Comment (Metaverse-Nabeel) [reply src/redux/categories/categoriesSlice.js]: Hello @GedeonTS 

Thank you for your kind feedback. I shall make the changes and ask for a review again.

Regards

**Review by brainconnect93 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @Metaverse-Nabeel, 👋 👋  

Your project is complete! Great job for following the project requirement🥇and implementing the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 




### To Highlight :1st_place_medal: 

- Descriptive Pull Request. :heavy_check_mark:
- Action and Reducer functionality. :heavy_check_mark:
- Ducks pattern file structure. :heavy_check_mark:
- `@reduxjs/toolkit` & `react-redux` present in package.json :heavy_check_mark:
- Linters environment properly setup :heavy_check_mark:
- Good use of Git flow :heavy_check_mark:

## Optional changes

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brainconnect93 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**